packages/vm/src/imports.rs

```diff
@@ -263,7 +264,16 @@ fn do_secp256k1_verify<A: BackendApi, S: Storage, Q: Querier>(
     let result = secp256k1_verify(&hash, &signature, &pubkey);
     let gas_info = GasInfo::with_cost(GAS_COST_VERIFY_SECP256K1_SIGNATURE);
     process_gas_info::<A, S, Q>(env, gas_info)?;
-    Ok(result?.into())
+    // Ok((!(result?)).into())
+    match result {
+        Ok(true) => Ok(0),
+        Ok(false) => Ok(1),
+        Err(CryptoError::HashErr { error_code, .. }) => Ok(error_code),
+        Err(CryptoError::SignatureErr { error_code, .. }) => Ok(error_code),
+        Err(CryptoError::PublicKeyErr { error_code, .. }) => Ok(error_code),
+        Err(CryptoError::GenericErr { error_code, .. }) => Ok(error_code),
+        Err(err) => panic!("unknown crypto error: {:?}", err),
```

Comment:
comwasm-crypto is internal only. So we can declare `CryptoError` to be a private type and rremove the `#[non_exhaustive]`. This protects us from runtime crashes when new cases are added

Reply:
You mean, through `pub(super)` or so?

Reply:
Through

```
#[docs(hidden)]
pub
```

But at the end of the day, the whole `cosmwasm-crypto` crate gets a big warning sign stating this is not for direct usage.